base/special/gamma.jl

```diff
@@ -33,49 +33,91 @@ Compute the logarithmic factorial of `x`
 lfact(x::Real) = (x<=1 ? zero(float(x)) : lgamma(x+one(x)))
 @vectorize_1arg Number lfact
 
-const clg_coeff = [76.18009172947146,
-                   -86.50532032941677,
-                   24.01409824083091,
-                   -1.231739572450155,
-                   0.1208650973866179e-2,
-                   -0.5395239384953e-5]
-
-function clgamma_lanczos(z)
-    const sqrt2pi = 2.5066282746310005
-
-    y = x = z
-    temp = x + 5.5
-    zz = log(temp)
-    zz = zz * (x+0.5)
-    temp -= zz
-    ser = complex(1.000000000190015, 0)
-    for j=1:6
-        y += 1.0
-        zz = clg_coeff[j]/y
-        ser += zz
-    end
-    zz = sqrt2pi*ser / x
-    return log(zz) - temp
-end
-
 """
     lgamma(x)
 
 Compute the logarithm of the absolute value of [`gamma`](:func:`gamma`) for
 [`Real`](:obj:`Real`) `x`, while for [`Complex`](:obj:`Complex`) `x` it computes the
-logarithm of `gamma(x)`.
+principal branch cut of the logarithm of `gamma(x)` (defined for negative `real(x)`
+by analytic continuation from positive `real(x)`).
 """
-function lgamma(z::Complex)
-    if real(z) <= 0.5
-        a = clgamma_lanczos(1-z)
-        b = log(sinpi(z))
-        const logpi = 1.14472988584940017
-        z = logpi - b - a
+function lgamma end
+
+# asymptotic series for log(gamma(z)), valid for sufficiently large real(z) or |imag(z)|
+@inline function lgamma_asymptotic(z::Complex{Float64})
+    zinv = inv(z)
+    t = zinv*zinv
+    # coefficients are bernoulli[2:n+1] .// (2*(1:n).*(2*(1:n) - 1))
+    return (z-0.5)*log(z) - z + 9.1893853320467274178032927e-01 + # <-- log(2pi)/2
+       zinv*@evalpoly(t, 8.3333333333333333333333368e-02,-2.7777777777777777777777776e-03,7.9365079365079365079365075e-04,-5.9523809523809523809523806e-04,8.4175084175084175084175104e-04,-1.9175269175269175269175262e-03,6.4102564102564102564102561e-03,-2.9550653594771241830065352e-02)
```

Comment:
I felt like it was actually easier to read the code when these lines were left unwrapped, especially on github or in an editor like Atom that does not line-wrap for you, because then you don't have several lines of indecipherable numbers interrupting the flow of the code.